### PR TITLE
 Update macosx framework packaging to follow apple guidelines

### PR DESCRIPTION
**Pull request by vraspar**
### Description
Refactor framework directory structure for MacOS packages

### Motivation and Context
Apple started enforcing specific [framework structure](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html) for MacOS packages. We need to change how we package for MacOS to follow the guidelines

### CODE REVIEW

File: tools/ios/build_xcframework.py [hunk 1]
```diff
@@ -213,19 +213,64 @@ def platform_arch_framework_build_dir(platform, arch):
...
-                    shutil.copytree(src, dst)
-                else:
-                    shutil.copy(src, dst)
-
-            # combine arch-specific framework libraries
-            arch_libs = [str(framework_dir / "onnxruntime_extensions") for framework_dir in arch_framework_dirs]
-            run(
-                *([_lipo, "-create", "-output", str(platform_fat_framework_dir / "onnxruntime_extensions")] + arch_libs)
-            )
+            # macos requires different framework structure:
```
Comment (skottmckay) [inline tools/ios/build_xcframework.py:216]: do we also need this for maccatalyst?

Comment (vraspar) [reply tools/ios/build_xcframework.py:216]: According to this [article](https://developer.apple.com/documentation/bundleresources/placing_content_in_a_bundle?language=objc), it seems that we also need the same structure for maccatalyst

File: .pipelines/templates/build-package-for-ios-cocoapods.yml [hunk 2]
```diff
@@ -173,13 +173,13 @@ jobs:
       POD_STAGING_DIR="$(Build.BinariesDirectory)/pod_staging"
       ARTIFACTS_STAGING_DIR="$(Build.ArtifactStagingDirectory)"
       POD_NAME="onnxruntime-extensions-c"
-      POD_ARCHIVE_BASENAME="pod-archive-${POD_NAME}-${ORT_EXTENSIONS_POD_VERSION}.zip"
+      POD_ARCHIVE_BASENAME="onnxruntime_extensions.xcframework.zip"
```
Comment (skottmckay) [inline .pipelines/templates/build-package-for-ios-cocoapods.yml:176]: nit: slight preference for keeping the version in the zip filename so we know what exactly is in it if someone downloads the artifact directly from a CI build. 

when we rename to include in the nuget package the version info is coming from nuget.

File: tools/ios/build_xcframework.py [hunk 3]
```diff
@@ -213,19 +212,61 @@ def platform_arch_framework_build_dir(platform, arch):
...
+            # https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html
+            if platform == "macosx" or platform == "maccatalyst":
```
Comment (skottmckay) [inline tools/ios/build_xcframework.py:217]: we used `macabi` in the ORT version. just checking that `maccatalyst` is correct here.

Comment (vraspar) [reply tools/ios/build_xcframework.py:217]: build_xcframework.py is slightly inconsistent with onnxruntime build xcframework script. It defines: 
```
_all_supported_platform_archs = {
    "iphoneos": ["arm64"],
    "iphonesimulator": ["x86_64", "arm64"],
    "macosx": ["x86_64", "arm64"],
    "maccatalyst": ["x86_64", "arm64"],
}
```

**Review by skottmckay — APPROVED** [review]

File: tools/ci_build/extract_nuget_files.ps1 [hunk 4]
```diff
@@ -44,6 +44,21 @@ Foreach-Object {
   Invoke-Expression -Command $cmd
 }
 
+# process iOS xcframework
+$xcframeworks = Get-ChildItem $Env:BUILD_BINARIESDIRECTORY\nuget-artifact -Filter onnxruntime_extensions.xcframework.*.zip
```
Comment (skottmckay) [inline tools/ci_build/extract_nuget_files.ps1:48]:
```suggestion
$xcframeworks = Get-ChildItem $Env:BUILD_BINARIESDIRECTORY\nuget-artifacts -Filter onnxruntime_extensions.xcframework.*.zip
```

File: tools/ci_build/extract_nuget_files.ps1 [hunk 5]
```diff
@@ -44,6 +44,21 @@ Foreach-Object {
...
+  $xcframework = $xcframeworks[0]
+  # remove version info from filename and use required filename format
+  $target_file = "$nuget_sources_dir\onnxruntime_extensions.xcframework.zip"
+  New-Item -Path $target_dir -ItemType directory
+
+  Write-Output "Copy-Item $($xcframework.FullName) $target_file"
+  Copy-Item $xcframework.FullName $target_file
+}
+elseif ($xcframeworks.Count -ne 1) {
+  Write-Error "Expected at most one onnxruntime_ios_xcframework*.zip file but got: [$xcframeworks]"
```
Comment (skottmckay) [inline tools/ci_build/extract_nuget_files.ps1:59]:
```suggestion
  Write-Error "Expected at most one onnxruntime_ios_xcframework.*.zip file but got: [$xcframeworks]"
```

File: tools/ci_build/extract_nuget_files.ps1 [hunk 6]
```diff
@@ -44,6 +44,21 @@ Foreach-Object {
...
+  $xcframework = $xcframeworks[0]
+  # remove version info from filename and use required filename format
+  $target_file = "$nuget_sources_dir\onnxruntime_extensions.xcframework.zip"
+  New-Item -Path $target_dir -ItemType directory
+
+  Write-Output "Copy-Item $($xcframework.FullName) $target_file"
+  Copy-Item $xcframework.FullName $target_file
+}
+elseif ($xcframeworks.Count -ne 1) {
+  Write-Error "Expected at most one onnxruntime_ios_xcframework.*.zip file but got: [$xcframeworks]"
```
Comment (skottmckay) [inline tools/ci_build/extract_nuget_files.ps1:59]:
```suggestion
  Write-Error "Expected at most one onnxruntime_extensions.xcframework.*.zip file but got: [$xcframeworks]"
```

**Review by skottmckay — APPROVED** [review]

**Pull request merged**